README.md

```diff
@@ -327,6 +327,49 @@ interface. In case of a missing `interface_name`, the `name` of the profile name
 **Note:** The `name` (the profile name) and the `interface_name` (the device name) may be
 different or the profile may not be tied to an interface at all.
 
+### `match`
+
+Settings to specify devices or systems matching a profile. Currently, only the `path`
+setting is implemented.
+
+The settings support a list of patterns which support the following modifiers and
+wildcards:
+
+**Special modifiers for `match` settings:**
+
+- `|`, the element is alternative, the match evaluates to be true if at least one of
+  the alternative element matches (logical OR). By default, an element is alternative.
+  This means that an element `foo` behaves the same as `|foo`
```

Comment:
This does not sound like correct English to me. Maybe it would be correct when using `alternative` as a noun - `the element is an alternative, ....` @richm how would you describe this?

Reply:
Suggested change
```
* `|` - logical OR - the match evaluates to be true if at least one of the elements on the left or right of the `|` matches.
* `&` - logical AND -  the match evaluates to be true if both the element on the left and the right of the `&` matches.
* `\` - escape special characters - can be used at the beginning of the element (after the optional
  special characters) to escape the start of the pattern. For example, `&\!a` is an
  mandatory match for literally `!a`

**Wildcard patterns for `match` Settings:**
In general these work like shell globs.

* `*` - matches zero or more of any character
* `?`- matches any single character
* `[fo]` - matches any single `f` or `o` character
```
However, I have some questions so that I can understand the intention of this functionality so that I can figure out how to describe it in English.

I do not understand this text "By default, an element is alternative.  This means that an element `foo` behaves the same as `|foo`"  - does that mean a single `path: foo` will match nothing *or* the string "foo"?

I also do not understand how `\` works - does it only work at the beginning of an element?  So I can't use `fo\!o` to match the literal string "fo!o"?

Does the `[...]` match support ranges?  e.g. `[a-z]` matches any lowercase character?

Reply:
Regarding the change, the logical operators are only valid at the beginning of the items, so a logical and would be:

```
- &example
- &bar
```

and not

```
- example&bar
```

so "left/right" does not make sense here.

Reply:
ok - so when it says "The settings support a list of patterns" - "list" here means a YAML `list` - that was not clear at all to me from reading the docs.  So each match element must be a single list element.

====

The value is a YAML list.  Each list element is a pattern.  The value is considered a match if ANY of the elements match.  For example:
```
match:
  path:
    - foo
    - bar
```
will match if the value matches "foo" *OR* "bar".  `path` elements also support the following modifiers and wildcards:

* `|` - logical OR - used at the beginning of the element - the match evaluates to be true if this element matches
* `&` - logical AND -  used at the beginning of the element - the match evaluates to be true if this element and all of the other elements match
* `!` - logical NOT - used at the beginning of the element, or following a `|` or `&` - the match evaluates to true if this element does not match - Note that `!foo` is a shortcut for `&!foo`
* `\` - escape special characters - used to escape the other special characters used for modifiers and wildcards, for when you need to match the literal string. For example, `\!a` is a match for the literal string `!a`.

**Wildcard patterns for `match` Settings:**
In general these work like shell globs.

* `*` - matches zero or more of any character
* `?`- matches any single character
* `[fo]` - matches any single `f` or `o` character - also supports ranges - `[0-9]` will match any single digit character
```